src/utils/sqs_consumer.ts

```diff
@@ -87,18 +78,15 @@ export class SqsConsumer {
         try {
             await this._handleMessage(message);
         } catch (err) {
-            logger.error(
-                {
-                    err,
-                    message,
-                    id: this._id,
-                },
-                'Encountered error while handling message',
-            );
-            // Retry message
-            await this._sqsClient.changeMessageVisibilityAsync(message.ReceiptHandle!, 0);
-            await delay(ONE_SECOND_MS);
-            return;
+            logger.error({ err, message, id: this._id }, 'Encountered error while handling message');
+
+            if (err instanceof SqsRetryableError) {
```

Comment:
Just double checking since I don't know JS. Can you confirm that `instanceOf` is good enough for our use case? I'm not too sure how classes work in JS (my understanding is that they modify the prototype chain, but that may be wrong)

Reply:
Yeah, that's absolutely correct - the prototype chain of this `err` object may look like:
- `err.__proto__.name === 'SqsRetryableError'`
-`err.__proto__.__proto__.name === 'Error'`

I believe `instanceof` will check the first one, and if it doesn't match, will continue up the `__proto__` chain (i.e. `__proto__.__proto__...`, until the root is hit (i.e. there is no more `__proto__` field), in which case it returns `false`. Of course, returns `true` if a match is found before that.